test/test-mnist.rb

```diff
@@ -0,0 +1,113 @@
+class MnistTest < Test::Unit::TestCase
+  include Helper::Sandbox
+
+  def setup_data
+    setup_sandbox
+
+    def @dataset.cache_dir_path
+      @cache_dir_path
+    end
+
+    def @dataset.cache_dir_path=(path)
+      @cache_dir_path = path
+    end
+    @dataset.cache_dir_path = @tmp_dir
+
+    def @dataset.download(output_path, url)
+
+      directory = (Pathname.new(__dir__) + "tmp").expand_path
+      image_paths = ["#{directory}/train-images-idx3-ubyte.gz",
+                     "#{directory}/t10k-images-idx3-ubyte.gz"]
+      label_paths = ["#{directory}/train-labels-idx1-ubyte.gz",
+                     "#{directory}/t10k-labels-idx1-ubyte.gz"]
+      image_num, image_size_x, image_size_y, label = 10, 28, 28, 1
+
+      Zlib::GzipWriter.open(output_path) do |gz|
+        if image_paths.include?(output_path.to_s)
```

Comment:
Can we use `output_path.basename.to_s.include?("-images-")` here?
It seems that `Pathname.new(__dir__) + "tmp"` is too implementation specific.

Reply:
I understood what you say, so I fixed it.